src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -486,6 +494,242 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     report_io();
 }
 
+void ov::npuw::CompiledModel::CompiledModelDesc::serialize(std::ostream& stream,
+                                                           const std::size_t& idx,
+                                                           const std::string& device) const {
+    using namespace ov::npuw::s11n;
+
+    LOG_DEBUG("Serializing CompiledModelDesc...");
+    LOG_BLOCK();
+
+    write(stream, device);
+
+    write(stream, replaced_by);
+    // FIXME: is this check enough?
+    if (replaced_by == idx) {
+        NPUW_ASSERT(compiled_model);
+        // FIXME: workaround for import/export model since import model seem to reset the file pointer
+        std::stringstream ss;
+        compiled_model->export_model(ss);
+        write(stream, ss.str());
+    }
+
+    write(stream, param_base);
+    write(stream, forced_to_fcall);
+
+    write(stream, host_gather.dst_idx);
+    write(stream, host_gather.src_idx);
+    write(stream, host_gather.idx_idx);
+
+    write(stream, spatial);
+
+    write(stream, scales);
+    write(stream, zerops);
+    write(stream, is_remote);
+
+    // NOTE: for closure only serialize uids - full flow
+    write(stream, closure_uid);
+
+    // Some tensors might be present in CPU closure already - need to serialize as is
+    // FIXME: When weightless serialization is introduced, this should be handled differently
+    write(stream, closure.size());
+    std::vector<ov::Tensor> cpu_closures;
+    std::vector<std::size_t> cpu_closure_ids;
+    for (std::size_t cidx = 0; cidx < closure.size(); ++cidx) {
+        if (closure_uid[cidx] == -1) {  // CPU closure, not in the bank
+            cpu_closure_ids.push_back(cidx);
+            cpu_closures.push_back(closure[cidx]);
+        }
+    }
+
+    write(stream, cpu_closure_ids);
+
+    for (const auto& tensor : cpu_closures) {
+        write(stream, tensor);
+    }
+
+    // FIXME: support weightless flow!
+
+    LOG_DEBUG("DONE.");
+}
+
+ov::npuw::CompiledModel::CompiledModelDesc ov::npuw::CompiledModel::CompiledModelDesc::deserialize(
+    std::istream& stream,
+    const std::size_t idx,
+    const std::shared_ptr<const ov::IPlugin>& plugin,
+    const ov::AnyMap& properties) {
+    using namespace ov::npuw::s11n;
+
+    LOG_DEBUG("Deserializing CompiledModelDesc...");
+    LOG_BLOCK();
+
+    CompiledModelDesc desc;
+
+    std::string device;
+    read(stream, device);
+    read(stream, desc.replaced_by);
+    // FIXME: is this check enough?
+    if (desc.replaced_by == idx) {
+        // Import model from either NPU or CPU plugin
+        // FIXME: workaround for import/export model since import model seems to reset the file pointer
+        std::string buf;
+        read(stream, buf);
+
+        // FIXME: extra copy
+        std::stringstream buffer;
+        buffer.write(&buf[0], buf.size());
+
+        // No NPUW properties are present in this config
+        desc.compiled_model = plugin->get_core()->import_model(buffer, device, properties);
+    }
```

Comment:
This is nearly correct.
`.replaced_by` may stay empty if this is a unique subgraph not turned into a function.

You'd better decided it via a blob you put to stream. You can put a binary flag ("has compiled model?") into stream first on serialization, so then you read this flag here again and decide what to do.

Reply:
Done